src/platform/datapath_raw.c

```diff
@@ -71,6 +125,30 @@ CxPlatDataPathInitialize(
     return Status;
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+VOID
+CxPlatDataPathRouteWorkerUninitialize(
+    _Inout_ CXPLAT_DATAPATH* DataPath
+    )
+{
+    CXPLAT_ROUTE_RESOLUTION_WORKER* Worker = DataPath->RouteResolutionWorker;
+
+    Worker->Enabled = FALSE;
+    CxPlatEventSet(Worker->Ready);
+    CxPlatEventWaitForever(Worker->Done);
```

Comment:
You don't need a separate event to wait on. You can wait on the thread itself.

Reply:
Right, some copy-paste stuff that needs clean-up.